test/common/http/dispatcher_test.cc

```diff
@@ -73,16 +73,18 @@ TEST_F(DispatcherTest, SetDestinationCluster) {
   callbacks_called cc = {0, 0, 0, 0, 0, 0};
   bridge_callbacks.context = &cc;
   bridge_callbacks.on_headers = [](envoy_headers c_headers, bool end_stream,
-                                   void* context) -> void {
-    ASSERT_TRUE(end_stream);
+                                   void* context) -> void* {
+    EXPECT_TRUE(end_stream);
     ResponseHeaderMapPtr response_headers = toResponseHeaders(c_headers);
     EXPECT_EQ(response_headers->Status()->value().getStringView(), "200");
     callbacks_called* cc = static_cast<callbacks_called*>(context);
     cc->on_headers_calls++;
+    return nullptr;
```

Comment:
Do you foresee the context being passed down all the way to the Http::Dispatcher or stopping in the bridge layer?

I ask because it would be great if instead of just updating the return type here we could actually cover the change with testing. But obviously the dispatcher is not seeing the context yet (or ever, hence my question).

Reply:
The dispatcher has no need to see it, since it doesn't utilize any signals returned by callbacks. So my inclination would be not to expose it at all there. My line of thinking was that the existing tests actually do cover the change, since they demonstrate that it causes no change in underlying functionality. Additional coverage will be needed for filters usage, of course.

Reply:
Just so I understand: we're returning a `nullptr` here since we don't expect the consumer (the dispatcher) to use the return value?

Reply:
> The dispatcher has no need to see it, since it doesn't utilize any signals returned by callbacks. So my inclination would be not to expose it at all there.

ok, that was my read of it.